plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -344,7 +344,13 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle, TupleDomain<ColumnHandle> tupleDomain)
     {
-        TableStatisticsCacheKey key = new TableStatisticsCacheKey(handle, tupleDomain);
+        return getTableStatistics(session, handle);
```

Comment:
`checkArgument(tupleDomain.isAll(), "Unexpected non-ALL constraint: %s", tupleDomain);`

(instead of ignoring silently)